server/src/main/scala/za/co/absa/atum/server/api/repository/CheckpointRepositoryImpl.scala

```diff
@@ -37,13 +44,63 @@ class CheckpointRepositoryImpl(writeCheckpointFn: WriteCheckpoint, writeCheckpoi
       "writeCheckpoint"
     )
   }
+
+  override def getCheckpointV2(partitioningId: Long, checkpointId: UUID): IO[DatabaseError, CheckpointV2DTO] = {
+    dbMultipleResultCallWithAggregatedStatus(
+      getCheckpointV2Fn(GetCheckpointV2Args(partitioningId, checkpointId)),
+      "getCheckpoint"
+    )
+      .map(_.flatten)
+      .flatMap { checkpointItems =>
+        ZIO
+          .fromEither(checkpointItemsToCheckpointV2DTO(checkpointItems))
+          .mapError(error => GeneralDatabaseError(error.getMessage))
+      }
+  }
+
+  private def checkpointItemsToCheckpointV2DTO(
```

Comment:
Why not to move this functionality directly to `GetPartitioningCheckpointV2.scala` ?

Reply:
I disagree. It's not a concern of the database function. It's simply a representation on Scala level of what happens on the DB level. The SQL code returns a certain data structure which is not CheckpointV2DTO. That's why we have repository layer in our application, we can do these mappings there. But if you feel we should move this particular method some place else then a companion of the CheckpointV2DTO could be one option. But I am not a big fan of that since CheckpointV2DTO is part of model module and I don't see a reason why should we expose this to agent module which depends on the model module. I think these so called DTOs should be used only at the boundaries of our application (for http communication). In other words at repository layer we should return CheckpointV2 and this class could have a method in its companion for conversion to DTO representation of this type. Unfortunately we haven't started doing it this way from the beginning but I would be happy to introduce this sort of convention of not mixing these data models. Let's discuss with others.